master/static/srv/enrich_task_logs.py

```diff
@@ -0,0 +1,95 @@
+import argparse
+import datetime
+import json
+import os
+import re
+import socket
+import sys
+import time
+from typing import Any, Dict
+
+from determined.common import api
+
+# Example log message given below.
+# 2022-05-12 16:32:48,757:gc_checkpoints: [rank=0] INFO: Determined checkpoint GC, version 0.17.16-dev0
+# Below regex is used to extract the rank field from the log message. 
+# Excluding empty spaces this regex matches rank in the above example as [rank=0]
+rank = re.compile("(?P<space1> ?)\[rank=(?P<rank_id>([0-9]+))\](?P<space2> ?)(?P<log>.*)")
+# Below regex is used to extract the message severity from the log message. 
+# Excluding empty spaces and delimiter(:) this regex matches message severity level in the above example as INFO
+level = re.compile("(?P<space1> ?)(?P<level>(DEBUG|INFO|WARNING|ERROR|CRITICAL)):(?P<space2> ?)(?P<log>.*)")
+
+
+def main(
+    master_url: str,
+    task_logging_metadata: Dict[str, Any],
+) -> None:
+    buffer = []
+    # Maximum number of log messages to be send to determined master each time.
+    max_buffer_length = 1024
+    start_time = time.time()
+    # Maximum time in seconds the loop executes, before sending log messages to determined master.
+    frequency = 5
```

Comment:
A 5-second buffer seems pretty high.  Certainly watching trial logs would be a jarring experience.  Is .5 sec too short?

Reply:
I'm just merging this code, it is only used by Slurm.    You should take it up with @stoksc because he directed the implementation :-)

Reply:
haha. I mean, for what it is worth I nearly insisted we work off https://github.com/determined-ai/determined-ee/pull/185/files but we took this approach instead.

Reply:
@rb-determined-ai can we fly with this for now, I'll try to land my batching over the next day?

Reply:
sure I guess.  Still working on one more view comment though.

Reply:
@rb-determined-ai https://github.com/determined-ai/determined-ee/pull/305/files I'll bring it back over once this lands